tyk_reverse_proxy_clone.go

```diff
@@ -238,7 +236,7 @@ func TykNewSingleHostReverseProxy(target *url.URL, spec *APISpec) *ReverseProxy
 
 		// No override, and no load balancing? Use the existing target
 		targetToUse := target
-		if switchTargets {
```

Comment:
There isn't a test to cover this functionality, it looks ok, but we can't guarantee it works (Use case is: a path that is being rewritten has a hostname in it, and so needs re-parsing so the proxy target changes)